### ISSUE TITLE
updating / changing deps

### ISSUE THREAD

**Issue by oriSomething** [toddmotto/html5blank#114]
since most of the deps in the project are out-dated, it's time to make some updates.
also i'm thinking of deleting `conditionizr` because it becoming more and more irrelevant. today using css features per browser is pretty much useless.
also i'm considering removing `modernizr` because, it's way more useful when you design for old browsers what also makes it useless currently. the only reason i'm thinking it's still useful for is `html5 shim/v`, but since `IE 8` stopped to be a priority for most devs, i'm not sure if there is a point of this bloating. maybe return using `IE conditional`s for `shim/v`s.

@toddmotto any thoughts about this?

steps:

updating `npm` deps:
- [ ] del
- [ ] gulp
- [ ] gulp-autoprefixer
- [ ] gulp-concat
- [ ] gulp-jshint
- [ ] gulp-livereload
- [ ] gulp-load-plugins
- [ ] gulp-template
- [ ] gulp-uglify
- [ ] jshint-stylish
- [ ] replace `gulp-ruby-sass` with `gulp-sass` for using node sass which way faster and doesn't require `ruby` installed

update `bower` deps:
- [ ] jQuery
- [ ] modernizr
- [ ] normalize.css

**Comment by ushka** [comment]
What about common/require JS for dependency mgmt.?

**Comment by oriSomething** [comment]
it's possible but... 
1. WordPress has their own dependencies management, and their using it for their stuff so it's a bit problematic
2. also there are deps you want to use in the `<head>` and not in `<body>`. the split make it complex
3. most WordPress sites, don't need much javascript if at all. (for example, my blog uses 10 lines of JS exclude `modernizr` and WordPress deps)
4. for most cases `requireJS` is more reasonable (after `ES6 modules` which poorly supported by transpilers). and it's dying. you'll need `CJS` with `browserify` / `webpack` plugins... which make it very complex

the bottom line, there is a lot of complexity involve in it, in a project that aim to be a ~~(HTML5)~~ **blank** canvas. it's something that belong to be another project which involve `React` or an `EmberCLI plugin theme`

**Comment by oriSomething** [comment]
@ushka maybe we should use [PHP-Parser](https://github.com/nikic/PHP-Parser) for build process. and maybe that way we can auto build `prod` and `dev` versions

**Comment by LeoFNaN** [comment]
I was looking at the gulp/npm dependencies and I was wondering whats the reason for keeping bower around?  You can pull in normalize.css, modernizr and jquery through npm.  That's one less dependency to maintain and run.  I just want to know if there is a reason to use it over npm.

**Comment by oriSomething** [comment]
@LeoFNaN it's right only if we move to `webpack` totally, and in this case it might happens. otherwise, `browserify` doesn't play as nice with `css` and using no bundler means, people won't be able to use most of `NPM` deps, since they're written solely to node.

but to admit, i'm not sure `webpack` is the right tool since, from what i see with WP developers, JavaScript tooling might be way too much for most, and configuring `webpack` is too demanding. Maybe, will create a solution such as `create-react-app` or simpler version of `ember-cli`. but i can't see this happens in the next three months since i'm too busy
